xcube_cmems/cmems.py

```diff
@@ -98,54 +116,94 @@ def get_opendap_urls(self, data_id) -> List[str]:
         return urls
 
     @staticmethod
-    def get_csw_records(csw, pagesize: int = 10, max_records: int = 300) \
-            -> Dict[Any, Any]:
-        """
-        Iterate max_records/pagesize times until the requested value in
-        max_records is reached.
-        return: Dictionary of CSW records objects
-        """
-        # Iterate over sorted results.
-        sortby = SortBy([SortProperty("dc:title", "ASC")])
-        csw_records = {}
-        start_position = 0
-        next_record = getattr(csw, "results", 1)
-        while next_record != 0:
-            csw.getrecords2(
-                startposition=start_position,
-                maxrecords=pagesize,
-                sortby=sortby,
-                esn='full'
-            )
-            csw_records.update(csw.records)
-            if csw.results["nextrecord"] == 0:
+    async def get_response(session: aiohttp.ClientSession,
+                           url: str,
+                           params: Dict) -> \
+            Optional[aiohttp.ClientResponse]:
+        num_retries = 10
+        for i in range(num_retries):
+            resp = await session.request(method='GET',
+                                         url=url,
+                                         ssl=True,
+                                         params=params)
+            if resp.status == 200:
+                return resp
+            elif resp.status == 429:
+                time.sleep(10)
+            else:
```

Comment:
Please consider relacing this naive retry using an exponential backoff. See xcube-sh or use library, e.g. `retry` package or similar.

Reply:
done